StortSpelProjekt/Engine/src/Renderer/Model.h

```diff
@@ -0,0 +1,48 @@
+#ifndef MODEL_H
+#define MODEL_H
+
+#include "Core.h"
+
+
+class Mesh;
+class Texture;
+class Resource;
+class ShaderResourceView;
+class DescriptorHeap;
+struct SlotInfo;
+
+// DX12 Forward Declarations
+struct ID3D12Device5;
+struct D3D12_INDEX_BUFFER_VIEW;
+
+class Model
+{
+public:
+    Model(const std::wstring path,
+        std::vector<Mesh*>* meshes,
+        std::vector<std::map<TEXTURE_TYPE, Texture*>>* textures);
+    virtual ~Model();
+
+    std::wstring GetPath() const;
+    unsigned int GetSize() const;
+
+    Mesh* GetMeshAt(unsigned int index);
+    std::map<TEXTURE_TYPE, Texture*> GetTexturesAt(unsigned int index);
+    SlotInfo GetSlotInfoAt(unsigned int index);
+
+    // Material
+    // TODO: maybe get/set when it's material
+    std::map<TEXTURE_TYPE, Texture*>* GetTextures(unsigned int index = 0);

```

Comment:
Possible duplicate of the function above?

Reply:
fixed